Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -59,4 +54,16 @@ private void HandleUserInterfaceRangeCheck(ref BoundUserInterfaceCheckRangeEvent
             }
         }
     }
+
+    public sealed class CanInteractWhileInsideContainerEvent : HandledEntityEventArgs
+    {
+        public readonly EntityUid User;
+        public readonly BaseContainer Container;
+
+        public CanInteractWhileInsideContainerEvent(EntityUid user, BaseContainer container)
+        {
+            User = user;
+            Container = container;
+        }
+    }
 }
```

Comment:
Please document your code as this is not self-descriptive at all why it exists.

Also all new events need to be structs raised by-ref unless you have good reason to otherwise.

Reply:
1984'ed for later PR maybe.